modules/solutions/twitter-analytics/widgets/Emotions/src/Emotions.jsx

```diff
@@ -55,21 +54,22 @@ class Emotions extends Widget {
         };
 
         this.providerConfig = {
-            type: 'RDBMSBatchDataProvider',
-            config: {
-                datasourceName: 'Twitter_Analytics',
-                queryData:{
-                    query: "select type as Sentiment, count(TweetID) as Rate from sentiment where PARSEDATETIME(timestamp, 'yyyy-mm-dd hh:mm:ss','en') > CURRENT_TIMESTAMP()-86400 group by type"
-                },
-                tableName: 'sentiment',
-                incrementalColumn: 'Sentiment',
-                publishingInterval: 60,
+            configs: {
```

Comment:
@GowthamyVaseekaran : we can move this to widgetconf.json file. So the config can be configurable for the user. Please check http-analytics or activity-explorer gadgets

Reply:
We were discussing updating the base widget to have this functionality(i.e get data provider conf in one function call). Shall we update this after that change is integrated to the base widget?

Reply:
@GowthamyVaseekaran Wso2-dashboard 1.2.1 is released, data provider confs can be moved to the widgetconf file easily now.

Reply:
+1